pkg/v1/v1util/zip.go

```diff
@@ -80,6 +84,27 @@ func GunzipReadCloser(r io.ReadCloser) (io.ReadCloser, error) {
 	}, nil
 }
 
+// EstargzReadCloser reads uncompressed input data from the io.ReadCloser and
+// returns an io.ReadCloser from which compressed data may be read.
+// Refer to estargz for the options:
+// https://pkg.go.dev/github.com/containerd/stargz-snapshotter@v0.2.0/estargz#Option
+func EstargzReadCloser(r io.ReadCloser, opts ...estargz.Option) (io.ReadCloser, v1.Hash, error) {
```

Comment:
Can we put this under `pkg/internal/estargz` while we're experimenting and expose it later? I'm not sure that I want to commit to this interface quite yet, and v1util is depended on by basically every package in this module, so it can't get pruned if it's in here.

Also `estargz.NewRead[Close?]er` reads nicely 😛

Reply:
This should be done, and I blew up the rest of v1util in: https://github.com/google/go-containerregistry/pull/872 for consistency.